src/nvim/fileio.c

```diff
@@ -5155,7 +5157,12 @@ int delete_recursive(char_u *name)
   int result = 0;
 
   if (os_isrealdir(name)) {
-    snprintf((char *)NameBuff, MAXPATHL, "%s/*", name);  // NOLINT
+    char_u *esc_name;
+    // Escape some regexp chars in file names as these probably will be
+    // interpreted as ill-formed regular expressions in gen_expand_wildcards.
+    esc_name = vim_strsave_escaped(name, regexp_escape_chars);
```

Comment:
I am wondering, what with `$`? It also cannot be escaped properly on Windows: vim/vim#541, 3.